src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -18,16 +18,25 @@ internal partial class SwitchBinder : LocalScopeBinder
         private TypeSymbol _switchGoverningType;
         private readonly GeneratedLabelSymbol _breakLabel;
 
-        private bool PatternsEnabled =>
-            (_switchSyntax?.SyntaxTree?.Options as CSharpParseOptions)?.IsFeatureEnabled(MessageID.IDS_FeaturePatternMatching) == true;
-
-        internal SwitchBinder(Binder next, SwitchStatementSyntax switchSyntax)
+        protected SwitchBinder(Binder next, SwitchStatementSyntax switchSyntax)
             : base(next)
         {
             _switchSyntax = switchSyntax;
             _breakLabel = new GeneratedLabelSymbol("break");
         }
 
+        internal static SwitchBinder Create(Binder next, SwitchStatementSyntax switchSyntax)
+        {
+            // When pattern-matching is enabled, we use a completely different binder and binding
+            // strategy for switch statements. Once we have confirmed that it is totally upward
+            // compatible with the existing syntax and semantics, we will remove *this* binder
+            // and use the new one for binding all switch statements.
```

Comment:
Since typeswitch is already in `master`, and this is an incremental improvement, I don't think a PROTYTPE comment is the best approach; there are already issues tracking this work. The plan to get confirmation is that all tests pass with the code in this source file totally removed.

---
In reply to: [64948711](https://github.com/dotnet/roslyn/pull/11485#discussion_r64948711) [](ancestors = 64948711,64946958)